Framework/Logger/include/Framework/Logger.h

```diff
@@ -71,10 +71,14 @@ inline FMT_ENABLE_IF_T(internal::is_string<S>::value, int)
 #endif
 FMT_END_NAMESPACE
 
-#define LOGF(severity, ...)                                                                                                                                        \
-  for (bool fairLOggerunLikelyvariable = false; fair::Logger::Logging(fair::Severity::severity) && !fairLOggerunLikelyvariable; fairLOggerunLikelyvariable = true) \
-  fmt::fprintf(fair::Logger(fair::Severity::severity, __FILE__, CONVERTTOSTRING(__LINE__), __FUNCTION__).Log(), __VA_ARGS__)
-#define LOGP(level, ...) LOG(level) << fmt::format(__VA_ARGS__)
+// clang-format off
+#define LOGF(severity, ...)                                                                                                                                          \
+  if (!fair::Logger::SuppressSeverity(fair::Severity::severity))                                                                                                     \
+    for (bool fairLOggerunLikelyvariable = false; fair::Logger::Logging(fair::Severity::severity) && !fairLOggerunLikelyvariable; fairLOggerunLikelyvariable = true) \
+      fmt::fprintf(fair::Logger(fair::Severity::severity, __FILE__, CONVERTTOSTRING(__LINE__), __FUNCTION__).Log(), __VA_ARGS__)
+// clang-format on
+#define LOGP(severity, ...) \
```

Comment:
I think braces would enhance the readability.
On a more fundamental level: If I understand correctly, this part of the code is compiled when we don't find the FairLogger header. How then can you use `fair::Logger::...` symbols?

Reply:
OK, in this case you are right. In the other parts actually we must not use braces, because
`LOG(level, "foo");` extents to (simplified)
```
if (level > debug) std::cout << "foo";
```
so there must not be a closing brace.

Reply:
Actually, at the second look, I don't want to add braces, since it would change the behavior due to the extra `;` ad the end.
Now, the `;` of the `LOG(...);` invocation becomes the closing semicolon. If I add braces, I'll have to add the semicolon inside, and there might be an extra semicolon thereafter.
This would break code like
```
if (foo)
  LOGP(debug, "foo");
else
  ...
```
(not, the test case doesn't have braces again..., but I hope it illustrates the problem. So I'd rather leave it as is.